apps/volontulo/models.py

```diff
@@ -38,13 +111,175 @@ class Offer(models.Model):
     benefits = models.TextField()
     location = models.CharField(max_length=150)
     title = models.CharField(max_length=150)
-    time_period = models.CharField(max_length=150)
-    status_old = models.CharField(max_length=30, default='NEW')
+    started_at = models.DateTimeField(blank=True, null=True)
+    finished_at = models.DateTimeField(blank=True, null=True)
+    time_period = models.CharField(max_length=150, default='', blank=True)
+    status_old = models.CharField(
+        max_length=30,
+        default='NEW',
+        null=True,
+        unique=False
+    )
+    offer_status = models.CharField(
+        max_length=16,
+        choices=OFFER_STATUSES,
+        default='unpublished',
+    )
+    recruitment_status = models.CharField(
+        max_length=16,
+        choices=RECRUITMENT_STATUSES,
+        default='open',
+    )
+    action_status = models.CharField(
+        max_length=16,
+        choices=ACTION_STATUSES,
+        default='ongoing',
+    )
     votes = models.BooleanField(default=0)
+    recruitment_start_date = models.DateTimeField(blank=True, null=True)
+    recruitment_end_date = models.DateTimeField(blank=True, null=True)
+    reserve_recruitment = models.BooleanField(blank=True, default=True)
+    reserve_recruitment_start_date = models.DateTimeField(
+        blank=True,
+        null=True
+    )
+    reserve_recruitment_end_date = models.DateTimeField(
+        blank=True,
+        null=True
+    )
+    action_ongoing = models.BooleanField(default=False, blank=True)
+    constant_coop = models.BooleanField(default=False, blank=True)
+    action_start_date = models.DateTimeField(blank=True, null=True)
+    action_end_date = models.DateTimeField(blank=True, null=True)
+    volunteers_limit = models.IntegerField(default=0, null=True, blank=True)
 
     def __str__(self):
         return self.title
 
+    def is_visible(self):
+        u"""Determine offer visibility for specified user type."""
+        statuses_keys = (
+            self.offer_status,
+            self.recruitment_status,
+            self.action_status
+        )
+        return True if self.VISIBILITY_MATRIX[statuses_keys] else False
```

Comment:
Couldn't self.VISIBILITY_MATRIX[status_key] be boolean (instead of 0 / 1), so that here it was
`return self.VISIBILITY_MATRIX`?